src/test/java/com/fasterxml/jackson/databind/deser/enums/EnumDeserializationTest.java

```diff
@@ -654,4 +654,34 @@ public void testIssue3006() throws Exception
         assertEquals(Operation3006.THREE, MAPPER.readValue("3", Operation3006.class));
         assertEquals(Operation3006.THREE, MAPPER.readValue(q("3"), Operation3006.class));
     }
+    
+   public void testEnumIndexAsKey() throws Exception
```

Comment:
the 3 space indents do not match up with the 4 space indents used - it seems in all of these files

Reply:
done~ my apologies for the confusion  😆